WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductRowViewModel.swift

```diff
@@ -63,6 +63,92 @@ final class ProductRowViewModel: ObservableObject, Identifiable {
     ///
     private let variationDisplayMode: VariationDisplayMode?
 
+    /// Subscription settings extracted from product meta data for a Subscription-type Product, if any
+    ///
+    private(set) var productSubscriptionDetails: ProductSubscription?
+
+    /// Description of the subscription billing details for a Subscription-type Product
+    /// eg: "$60.00 / 2 months"
+    ///
+    var subscriptionBillingDetailsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        guard let subscriptionPrice = productSubscriptionDetails?.price,
+              let subscriptionInterval = productSubscriptionDetails?.periodInterval,
+              let subscriptionPeriod = productSubscriptionDetails?.period,
+              let formattedPrice = currencyFormatter.formatAmount(subscriptionPrice, with: currency) else {
+            return ""
+        }
+
+        let subscriptionFrequency = {
+            switch subscriptionInterval {
+            case "1":
+                return subscriptionPeriod.descriptionSingular
+            default:
+                return subscriptionPeriod.descriptionPlural
+            }
+        }
+        return String.localizedStringWithFormat(Localization.formattedProductSubscriptionBilling,
+                                                formattedPrice, subscriptionInterval, subscriptionFrequency())
```

Comment:
Nit: You can define `subscriptionFrequency` as a computed constant instead of a closure for readability:

```suggestion
        let subscriptionFrequency = {
            switch subscriptionInterval {
            case "1":
                return subscriptionPeriod.descriptionSingular
            default:
                return subscriptionPeriod.descriptionPlural
            }
        }()
        return String.localizedStringWithFormat(Localization.formattedProductSubscriptionBilling,
                                                formattedPrice, subscriptionInterval, subscriptionFrequency)
```

Reply:
Oh! Definitely looks better when calling the formatting method. Updated: e92b901266ba7dadfb93b4a88a31da7a00641486